src/lambdas/api/app.js

```diff
@@ -0,0 +1,187 @@
+const satlib = require('../../lib')
+const api = require('../../lib/api')
+const { readYaml } = require('../../lib/fs')
+
+/**
+ * @typedef {import('express').Request} Request
+ * @typedef {import('express').Response} Response
+ * @typedef {import('express').NextFunction} NextFunction
+ * @typedef {import('express').ErrorRequestHandler} ErrorRequestHandler
+ */
+
+/**
+ * @param {Request} req
+ * @returns {string}
+ */
+const determineEndpoint = (req) => {
+  if (process.env['STAC_API_URL']) return process.env['STAC_API_URL']
+
+  if (req.get('X-Forwarded-Host')) {
+    return `${req.get('X-Forwarded-Proto')}://${req.get('X-Forwarded-Host')}`
+  }
+
+  return req.event && req.event.requestContext && req.event.requestContext.stage
+    ? `${req.get('X-Forwarded-Proto')}://${req.get('Host')}/${req.event.requestContext.stage}`
+    : `${req.get('X-Forwarded-Proto')}://${req.get('Host')}`
+}
+
+/**
+ * @param {Request} req
+ * @param {Response} _res
+ * @param {NextFunction} next
+ * @returns {void}
+ */
+const addEndpointToRequest = (req, _res, next) => {
+  req.endpoint = determineEndpoint(req)
+  next()
+}
+
+const app = express()
+
+app.use(logger('dev'))
+app.use(cors())
+app.use(express.json())
+app.use(addEndpointToRequest)
+
+app.get('/', async (req, res, next) => {
+  try {
+    res.json(await api.getCatalog(satlib.es, req.endpoint))
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/api', async (_req, res, next) => {
+  try {
+    const spec = await readYaml(path.join(__dirname, 'api.yaml'))
+    res.json(spec)
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/conformance', async (_req, res, next) => {
+  try {
+    res.json(await api.getConformance())
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/search', async (req, res, next) => {
+  try {
+    res.json(await api.searchItems(null, req.query, satlib.es, req.endpoint, 'GET'))
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.post('/search', async (req, res, next) => {
+  try {
+    res.json(await api.searchItems(null, req.body, satlib.es, req.endpoint, 'POST'))
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/collections', async (req, res, next) => {
+  try {
+    res.json(await api.getCollections(satlib.es, req.endpoint))
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/collections/:collectionId', async (req, res, next) => {
+  try {
+    const response = await api.getCollection(req.params.collectionId, satlib.es, req.endpoint)
+
+    if (response instanceof Error) next(createError(404))
+    else res.json(response)
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.get('/collections/:collectionId/items', async (req, res, next) => {
+  try {
+    res.json(await api.searchItems(
+      req.params.collectionId,
+      req.query,
+      satlib.es,
+      req.endpoint,
+      'GET'
+    ))
+  } catch (error) {
+    next(error)
+  }
+})
+
+app.post('/collections/:collectionId/items', async (req, res, next) => {
```

Comment:
just curious if you added this route or if this was the behavior from before? b/c POST to this endpoint is actually supposed to create a new item rather than do a query per the transactions extension. We'll be fixing this soon anyway, so no reason to change it now.

Reply:
That search capability existed before this PR

https://github.com/stac-utils/stac-server/blob/main/tests/system/test-api.js#L39